<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>FeatureDetectionParamsControlWidget</class> 
 <widget class="QWidget" name="FeatureDetectionParamsControlWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>300</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>FeatureDetectionParams</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <property name="margin">
    <number>3</number>
   </property>
   <property name="spacing">
    <number>3</number>
   </property>
    <item row="0" column="0">
     <widget class="QLabel" name="label_0">
       <property name="text">
         <string>detector</string>
       </property>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QLineEdit" name="detectorEdit">
       <property name="toolTip">
         <string>detector</string>
       </property>
       <property name="text">
           <string>ORB</string>
       </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QLabel" name="label_1">
       <property name="text">
         <string>descriptor</string>
       </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QLineEdit" name="descriptorEdit">
       <property name="toolTip">
         <string>descriptor</string>
       </property>
       <property name="text">
           <string>ORB</string>
       </property>
     </widget>
    </item>
    <item row="2" column="0">
     <widget class="QLabel" name="label_2">
       <property name="text">
         <string>matcher</string>
       </property>
     </widget>
    </item>
    <item row="2" column="1">
     <widget class="QLineEdit" name="matcherEdit">
       <property name="toolTip">
         <string>matcher</string>
       </property>
       <property name="text">
           <string>BF</string>
       </property>
     </widget>
    </item>
    <item row="3" column="0">
     <widget class="QLabel" name="label_3">
       <property name="text">
         <string>b2bThreshold</string>
       </property>
     </widget>
    </item>
    <item row="3" column="1">
     <widget class="QDoubleSpinBox" name="b2bThresholdSpinBox">
       <property name="toolTip">
         <string>b2bThreshold</string>
       </property>
       <property name="decimals">
        <number>6</number>
       </property>
       <property name="minimum">
        <double>0</double>
       </property>
       <property name="maximum">
        <double>256</double>
       </property>
       <property name="singleStep">
        <double>0.01</double>
       </property>
       <property name="value">
        <double>0.9</double>
       </property>
       <property name="prefix">
        <string></string>
       </property>
       <property name="suffix">
        <string></string>
       </property>
     </widget>
    </item>
    <item row="4" column="0">
     <widget class="QLabel" name="label_4">
       <property name="text">
         <string>matchF2F</string>
       </property>
     </widget>
    </item>
    <item row="4" column="1">
     <widget class="QCheckBox" name="matchF2FCheckBox">
       <property name="toolTip">
         <string>matchF2F</string>
       </property>
       <property name="checked">
           <bool>0</bool>
       </property>
     </widget>
    </item>
    <item row="5" column="0">
     <widget class="QLabel" name="label_5">
       <property name="text">
         <string>parameters</string>
       </property>
     </widget>
    </item>
    <item row="5" column="1">
     <widget class="QLineEdit" name="parametersEdit">
       <property name="toolTip">
         <string>Additional parameters</string>
       </property>
       <property name="text">
           <string></string>
       </property>
     </widget>
    </item>
    <item row="6" column="0">
     <widget class="QLabel" name="label_6">
       <property name="text">
         <string>plotTracks</string>
       </property>
     </widget>
    </item>
    <item row="6" column="1">
     <widget class="QCheckBox" name="plotTracksCheckBox">
       <property name="toolTip">
         <string>Draw tracks on source images</string>
       </property>
       <property name="checked">
           <bool>0</bool>
       </property>
     </widget>
    </item>
    <item row="7" column="0">
     <widget class="QLabel" name="label_7">
       <property name="text">
         <string>thresholdDistance</string>
       </property>
     </widget>
    </item>
    <item row="7" column="1">
     <widget class="QCheckBox" name="thresholdDistanceCheckBox">
       <property name="toolTip">
         <string>Threshold by descriptor distance instead of b2b ratio</string>
       </property>
       <property name="checked">
           <bool>0</bool>
       </property>
     </widget>
    </item>
    <item row="8" column="0">
     <widget class="QLabel" name="label_8">
       <property name="text">
         <string>maxFeatureCount</string>
       </property>
     </widget>
    </item>
    <item row="8" column="1">
     <widget class="QSpinBox" name="maxFeatureCountSpinBox">
       <property name="toolTip">
         <string>Max acceptable feature point count</string>
       </property>
       <property name="minimum">
        <number>0</number>
       </property>
       <property name="maximum">
        <number>32768</number>
       </property>
       <property name="singleStep">
        <number>1</number>
       </property>
       <property name="value">
        <number>4000</number>
       </property>
     </widget>
    </item>
   <item row="9" column="0">
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>0</width>
       <height>0</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="9" column="2">
    <spacer name="horizontalSpacer">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>0</width>
       <height>0</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="3" margin="3"/>
 <resources/>
 <connections>
  <connection>
   <sender>detectorEdit</sender>
   <signal>textChanged(QString)</signal>
   <receiver>FeatureDetectionParamsControlWidget</receiver>
   <slot>changeParameters()</slot>
  </connection>
  <connection>
   <sender>descriptorEdit</sender>
   <signal>textChanged(QString)</signal>
   <receiver>FeatureDetectionParamsControlWidget</receiver>
   <slot>changeParameters()</slot>
  </connection>
  <connection>
   <sender>matcherEdit</sender>
   <signal>textChanged(QString)</signal>
   <receiver>FeatureDetectionParamsControlWidget</receiver>
   <slot>changeParameters()</slot>
  </connection>
  <connection>
   <sender>b2bThresholdSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>FeatureDetectionParamsControlWidget</receiver>
   <slot>changeParameters()</slot>
  </connection>
  <connection>
   <sender>matchF2FCheckBox</sender>
   <signal>stateChanged(int)</signal>
   <receiver>FeatureDetectionParamsControlWidget</receiver>
   <slot>changeParameters()</slot>
  </connection>
  <connection>
   <sender>parametersEdit</sender>
   <signal>textChanged(QString)</signal>
   <receiver>FeatureDetectionParamsControlWidget</receiver>
   <slot>changeParameters()</slot>
  </connection>
  <connection>
   <sender>plotTracksCheckBox</sender>
   <signal>stateChanged(int)</signal>
   <receiver>FeatureDetectionParamsControlWidget</receiver>
   <slot>changeParameters()</slot>
  </connection>
  <connection>
   <sender>thresholdDistanceCheckBox</sender>
   <signal>stateChanged(int)</signal>
   <receiver>FeatureDetectionParamsControlWidget</receiver>
   <slot>changeParameters()</slot>
  </connection>
  <connection>
   <sender>maxFeatureCountSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>FeatureDetectionParamsControlWidget</receiver>
   <slot>changeParameters()</slot>
  </connection>
 </connections>
 <slots>
  <slot>changeParameters()</slot>
 </slots>
 <customwidgets>
 </customwidgets>
</ui>
